### PR TITLE
Graphclust tools

### PR DESCRIPTION
**Pull request by eteriSokhoyan**
Tools for graphclust pipeline.

### CODE REVIEW

**Comment by mmiladi** [comment]
@eteriSokhoyan Great! we are making progress.

**Comment by mmiladi** [comment]
@eteriSokhoyan please have a look at below, one can use has_text and has_n_columns for a flexible comparison:
https://docs.galaxyproject.org/en/master/dev/schema.html#tool-tests-test-output-assert-contents

Also we only need to test for main output files and values, i.e. cluster and partition files. Checking variables such as floay/numbers min-overlap,... is not needed

**Comment by mmiladi** [comment]
Hi @eteriSokhoyan @bgruening , congrats it's passing!
I looked into the .pl files, they follow original GraphClust style and look good for me. Just please consider a few things to update:

- [ ] Remove consecutive empty lines: `CollectResults/gc_res.pl `
- [ ] Use warnings: `PrepareForMlocarna/rnaclustScores2Dist.pl` 
- [ ] Rename directory: `GraphClust/mLocaRna/` to `GraphClust/LocARNAGraphClust/`
- [ ] Use warnings: `mLocaRna/foldFasta.pl` 
- [ ] Rename file: `mLocaRna/newMloc.pl` 
- [ ] Use warnings: `GraphClust/mLocaRna/scoreAln.pl `

**Comment by mmiladi** [comment]
Yes, the file is touched when the step is succeeded.

File: tools/GraphClust/LocARNAGraphClust/mlocarna.xml [hunk 1]
```diff
@@ -0,0 +1,189 @@
+<tool id="mlocarna" name="mlocarna" version="0.1.0" >
+	<requirements>
```
Comment (bgruening) [inline tools/GraphClust/LocARNAGraphClust/mlocarna.xml]: @mmiladi is this wrapper really needed or can we take the one from Sebastian?

**Comment by bgruening** [comment]
more empty files:  tools/GraphClust/NSPDK/test-data/SVECTOR/data.svector.1.fast_cluster
 tools/GraphClust/NSPDK/test-data/SVECTOR/data.svector.1.fast_cluster_sim
tools/GraphClust/NSPDK/test-data/SVECTOR/data.svector.blacklist.1
tools/GraphClust/NSPDK/test-data/SVECTOR/data.svector.fast_cluster.1.DONE

File: tools/GraphClust/LocARNAGraphClust/mlocarna.xml (hunk 1 above)
Comment (mmiladi) [inline tools/GraphClust/LocARNAGraphClust/mlocarna.xml]: @bgruening It's a wrapper that invokes locarna as one of the steps and pre/post-process input/output and creates custom files by reading tree, etc. So more or less needed. I would suggest @eteriSokhoyan to rename it something like: locarna_best_subtree ..

**Comment by mmiladi** [comment]
> more empty files:

Yes. *.DONE file are always created by successful run. The other files are empty for this test case but in general may not be empty,

**Comment by mmiladi** [comment]
@bgruening About the license file should we directly put the GPLv3 file inside the `tools/GraphClust` directory? https://github.com/BackofenLab/GraphClust/blob/master/COPYING

**Comment by bgruening** [comment]
Not needed in Galaxy wrappers. But this is an interesting discussion, Oo, I will drop by in your office and we can discuss this. Thanks a lot @eteriSokhoyan for fixing the last bit!

**Comment by mmiladi** [comment]
Good, thanks for your helps.

**Comment by bgruening** [comment]
@eteriSokhoyan ideally you should be able to `git rm *.pl` files and add this package to your requirements: https://anaconda.org/bioconda/graphclust-wrappers

**Comment by mmiladi** [comment]
@eteriSokhoyan Also care that all invocations in xml or other codes of form: './code.pl' must be changed to 'code.pl'

**Comment by mmiladi** [comment]
Thanks much both for your efforts. Could you please have a final look and eventually merge :-D ?

File: tools/GraphClust/Preprocessing/test-data/FASTA/orig.fasta [hunk 2]
```diff
@@ -0,0 +1,20 @@
+>RF00001_rep.0_AL096764.11/46123-46004_1 RF00001_rep.0
```
Comment (bgruening) [inline tools/GraphClust/Preprocessing/test-data/FASTA/orig.fasta]: This is also not used, isn't it? It is not mentioned in the XML files.

Comment (mmiladi) [inline tools/GraphClust/Preprocessing/test-data/FASTA/orig.fasta]: In compact: it is safe to delete for testing the  xml.

**Comment by bgruening** [comment]
We have it!
Thanks a lot - this is awesome! I think it was worth the last effort, this is now like it should be! 

Thanks again!

**Comment by mmiladi** [comment]
Wonderful! Thank you!
In fact now I see that was a very good idea to seperate xml from the code.
Would this be automatically added to the Freiburg instance?

**Comment by bgruening** [comment]
No. This is now at the TS. From there I need to pull it into the Freiburg Instance, but only after the workshop :) I will keep you posted.

**Pull request merged**